### PR TITLE
Spp tactical integration

### PR DESCRIPTION
**Pull request by akssinhaamit**
# Motivation and Context

# What has changed

# How to test?

# Links

### CODE REVIEW

**Review by insacuri — COMMENTED** [review]
I didn't do a full review sorry, but there are few things I saw.

File: _infra/docker/Dockerfile [hunk 1]
```diff
@@ -8,5 +8,5 @@ WORKDIR /app
 COPY . /app
 RUN pipenv install --deploy --system
 
-CMD ["gunicorn", "-b", "0.0.0.0:8080", "rm_reporting:app", "-w=4"]
+CMD ["gunicorn", "-b", "0.0.0.0:8080", "rm_reporting:app"]
```
Comment (warrenbailey) [inline _infra/docker/Dockerfile:11]: Doesn't this remove the workers?

**Review by warrenbailey — CHANGES_REQUESTED** [review]

File: _infra/helm/reporting/values.yaml [hunk 2]
```diff
@@ -37,6 +37,15 @@ database:
...
\ No newline at end of file
+crons:
+  sppReportScheduler:
+    name: rm-reporting-spp-reporting-scheduler
+    cron: "0 23 * * *"
+    target: "spp-reporting-api/v1/spp-reporting/send-report"
+
+gcp:
+  project: ras-rm-sandbox
+  bucket: ras-rm-notify-test
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply _infra/helm/reporting/values.yaml:48]: I will provide the dev as the default value although it gets governed by config

File: rm_reporting/resources/response_chasing.py [hunk 3]
```diff
@@ -82,7 +82,7 @@ def get(collection_exercise_id, survey_id):
                         "LEFT JOIN respondent_details rd ON bd.business_party_uuid = rd.business_party_uuid " \
                         "ORDER BY sampleunitref, case_status;"
         try:
-            collex_details = engine.execute(text(collex_status))
+            collex_details = engine.execute(text(collex_status)).first()
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply rm_reporting/resources/response_chasing.py:85]: 👍

File: _infra/docker/Dockerfile (hunk 1 above)
Comment (akssinhaamit) [reply _infra/docker/Dockerfile:11]: 👍

File: _infra/helm/reporting/templates/deployment.yaml [hunk 4]
```diff
@@ -133,5 +133,31 @@ spec:
...
             {{- end }}
+          - name: GOOGLE_CLOUD_PROJECT
+            value: "{{ .Values.gcp.project }}"
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: /var/secrets/google/credentials.json
+          - name: GCS_BUCKET_NAME
+            value: "{{ .Values.gcp.bucket }}"
+          - name: AWS_ENABLED
+            value: "{{ .Values.aws.enabled }}"
+          {{- if .Values.aws.enabled }}
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply _infra/helm/reporting/templates/deployment.yaml:150]: Sure.. in both the cases we (RAS/RM) won't be owning AWS account and I believe dev cluster will not like to communicate to AWS services. Hence keeping AWS enabled seems ok to me :-)

File: _infra/helm/reporting/templates/deployment.yaml [hunk 5]
```diff
@@ -133,5 +133,31 @@ spec:
...
+            valueFrom:
+              secretKeyRef:
+                name: aws-credentials
+                key: aws-key
+          {{- end }}
+          {{- if .Values.aws.enabled }}
+          - name: S3_SECRET_KEY
+            valueFrom:
+              secretKeyRef:
+                name: aws-credentials
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply _infra/helm/reporting/templates/deployment.yaml:165]: sure, but we can change it anytime we introduce another account in future.

**Review by warrenbailey — CHANGES_REQUESTED** [review]

File: rm_reporting/resources/spp_reporting.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+        respondent_details = engine.execute(text(respondent_query))
+    except SQLAlchemyError:
+        logger.exception("SQL Alchemy query failed")
+        raise
+    return respondent_details
+
+
+def get_collection_exercise_and_survey(engine):
+    collex_query = text(
+        'SELECT c.survey_uuid AS surveyId, c.id as collectionExerciseId '
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply rm_reporting/resources/spp_reporting.py:61]: 👍

File: rm_reporting/resources/spp_reporting.py [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+            'email': row[3],
+            'accountStatus': row[4],
+            'enrolmentStatus': row[5]
+        }
+        respondents.append(respondent)
+    return respondents
+
+
+def get_respondent_details(business_party_id, engine, survey_id):
+    respondent_query = "SELECT r.id, " \
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply rm_reporting/resources/spp_reporting.py:38]: Having respondent with the same query will duplicate the data on case level and then we will need to have additional logic to sort it out hence it was kept as a seperate step

File: rm_reporting/resources/spp_reporting.py [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+        case_details = get_case_details(survey_id, collection_exercise_id, engine)
+        survey_response_status = {'surveyId': survey_id, 'collectionExerciseId': collection_exercise_id}
+        reporting_unit_respondent_information = {'surveyId': survey_id}
+        populate_json_record(
+            survey_id,
+            case_details,
+            survey_response_status,
+            reporting_unit_respondent_information,
+            engine)
+        day = datetime.today().strftime('%d%m%Y')
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply rm_reporting/resources/spp_reporting.py:175]: 👍

File: rm_reporting/resources/spp_reporting.py [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+        'SELECT c.survey_uuid AS surveyId, c.id as collectionExerciseId '
+        'FROM collectionexercise.collectionexercise c '
+        'where survey_uuid = (select id from survey.survey where shortname = \'RSI\') '
+        'AND statefk = \'LIVE\''
+    )
+
+    collex_survey = engine.execute(collex_query).first()
+
+    if any(column is None for column in collex_survey.values()):
+        raise NoDataException
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply rm_reporting/resources/spp_reporting.py]: 👍

File: rm_reporting/resources/spp_reporting.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (warrenbailey) [reply rm_reporting/resources/spp_reporting.py:38]: as discussed happy with this now 👍

**Review by warrenbailey — APPROVED** [review]

**Comment by warrenbailey** [comment]
LGTM just need to sort the lint error 👍

**Pull request merged**